### ISSUE TITLE
Composite Primary Key Results in Wrong Additional Relationship Being Generated

### ISSUE THREAD

**Issue by alexanderdavide** [sequelize/sequelize-auto#525]
## Apparent bug
I have a composite primary key (see attatched database model) which results in an additional relationship being generated which is wrong on the one hand and triggers an `duplicate identifier` error on the other hand.

## Environment
sequelize: `6.6.2`
sequelize-auto: `0.8.2`

## Description
Kindly, review the attatched database model. The composite primary key on table `Measurement` is the source of the bug. When having it defined like on the database model, sequelize-auto generates wrong models for `DataSource` and `Location` while the `Measurement` model seems correct. See the code below; I marked the wrong parts with '### THIS IS WRONG ###'. If I exclude `LocationId` from the composite primary key, these parts are no longer generated so that the result is correct.

Location:
```ts
import * as Sequelize from 'sequelize';
import { DataTypes, Model, Optional } from 'sequelize';
import type { DataSource, DataSourceId } from './DataSource';
import type { Measurement, MeasurementId } from './Measurement';

export interface LocationAttributes {
  LocationId?: number;
  LocationUri: string;
}

export type LocationPk = "LocationId";
export type LocationId = Location[LocationPk];
export type LocationCreationAttributes = Optional<LocationAttributes, LocationPk>;

export class Location extends Model<LocationAttributes, LocationCreationAttributes> implements LocationAttributes {
  LocationId?: number;
  LocationUri!: string;

  // Location hasMany DataSource via LocationId
  DataSources!: DataSource[];
  getDataSources!: Sequelize.HasManyGetAssociationsMixin<DataSource>;
  setDataSources!: Sequelize.HasManySetAssociationsMixin<DataSource, DataSourceId>;
  addDataSource!: Sequelize.HasManyAddAssociationMixin<DataSource, DataSourceId>;
  addDataSources!: Sequelize.HasManyAddAssociationsMixin<DataSource, DataSourceId>;
  createDataSource!: Sequelize.HasManyCreateAssociationMixin<DataSource>;
  removeDataSource!: Sequelize.HasManyRemoveAssociationMixin<DataSource, DataSourceId>;
  removeDataSources!: Sequelize.HasManyRemoveAssociationsMixin<DataSource, DataSourceId>;
  hasDataSource!: Sequelize.HasManyHasAssociationMixin<DataSource, DataSourceId>;
  hasDataSources!: Sequelize.HasManyHasAssociationsMixin<DataSource, DataSourceId>;
  countDataSources!: Sequelize.HasManyCountAssociationsMixin;
  // Location belongsToMany DataSource via LocationId and DataSourceId  ### THIS IS WRONG ###
  DataSources!: DataSource[];
  getDataSources!: Sequelize.BelongsToManyGetAssociationsMixin<DataSource>;
  setDataSources!: Sequelize.BelongsToManySetAssociationsMixin<DataSource, DataSourceId>;
  addDataSource!: Sequelize.BelongsToManyAddAssociationMixin<DataSource, DataSourceId>;
  addDataSources!: Sequelize.BelongsToManyAddAssociationsMixin<DataSource, DataSourceId>;
  createDataSource!: Sequelize.BelongsToManyCreateAssociationMixin<DataSource>;
  removeDataSource!: Sequelize.BelongsToManyRemoveAssociationMixin<DataSource, DataSourceId>;
  removeDataSources!: Sequelize.BelongsToManyRemoveAssociationsMixin<DataSource, DataSourceId>;
  hasDataSource!: Sequelize.BelongsToManyHasAssociationMixin<DataSource, DataSourceId>;
  hasDataSources!: Sequelize.BelongsToManyHasAssociationsMixin<DataSource, DataSourceId>;
  countDataSources!: Sequelize.BelongsToManyCountAssociationsMixin;
  // Location hasMany Measurement via LocationId
  Measurements!: Measurement[];
  getMeasurements!: Sequelize.HasManyGetAssociationsMixin<Measurement>;
  setMeasurements!: Sequelize.HasManySetAssociationsMixin<Measurement, MeasurementId>;
  addMeasurement!: Sequelize.HasManyAddAssociationMixin<Measurement, MeasurementId>;
  addMeasurements!: Sequelize.HasManyAddAssociationsMixin<Measurement, MeasurementId>;
  createMeasurement!: Sequelize.HasManyCreateAssociationMixin<Measurement>;
  removeMeasurement!: Sequelize.HasManyRemoveAssociationMixin<Measurement, MeasurementId>;
  removeMeasurements!: Sequelize.HasManyRemoveAssociationsMixin<Measurement, MeasurementId>;
  hasMeasurement!: Sequelize.HasManyHasAssociationMixin<Measurement, MeasurementId>;
  hasMeasurements!: Sequelize.HasManyHasAssociationsMixin<Measurement, MeasurementId>;
  countMeasurements!: Sequelize.HasManyCountAssociationsMixin;

  static initModel(sequelize: Sequelize.Sequelize): typeof Location {
    Location.init({
    LocationId: {
      autoIncrement: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    LocationUri: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: "Location_LocationUri_key"
    }
  }, {
    sequelize,
    tableName: 'Location',
    schema: 'public',
    timestamps: false,
    indexes: [
      {
        name: "Location_LocationUri_key",
        unique: true,
        fields: [
          { name: "LocationUri" },
        ]
      },
      {
        name: "Location_pkey",
        unique: true,
        fields: [
          { name: "LocationId" },
        ]
      },
    ]
  });
  return Location;
  }
}
```

DataSource:
```ts
import * as Sequelize from 'sequelize';
import { DataTypes, Model, Optional } from 'sequelize';
import type { Location, LocationId } from './Location';
import type { Measurement, MeasurementId } from './Measurement';

export interface DataSourceAttributes {
  DataSourceId?: number;
  DataSourceUri: string;
  LocationId: number;
}

export type DataSourcePk = "DataSourceId";
export type DataSourceId = DataSource[DataSourcePk];
export type DataSourceCreationAttributes = Optional<DataSourceAttributes, DataSourcePk>;

export class DataSource extends Model<DataSourceAttributes, DataSourceCreationAttributes> implements DataSourceAttributes {
  DataSourceId?: number;
  DataSourceUri!: string;
  LocationId!: number;

  // DataSource belongsToMany Location via DataSourceId and LocationId ### THIS IS WRONG ###
  Locations!: Location[];
  getLocations!: Sequelize.BelongsToManyGetAssociationsMixin<Location>;
  setLocations!: Sequelize.BelongsToManySetAssociationsMixin<Location, LocationId>;
  addLocation!: Sequelize.BelongsToManyAddAssociationMixin<Location, LocationId>;
  addLocations!: Sequelize.BelongsToManyAddAssociationsMixin<Location, LocationId>;
  createLocation!: Sequelize.BelongsToManyCreateAssociationMixin<Location>;
  removeLocation!: Sequelize.BelongsToManyRemoveAssociationMixin<Location, LocationId>;
  removeLocations!: Sequelize.BelongsToManyRemoveAssociationsMixin<Location, LocationId>;
  hasLocation!: Sequelize.BelongsToManyHasAssociationMixin<Location, LocationId>;
  hasLocations!: Sequelize.BelongsToManyHasAssociationsMixin<Location, LocationId>;
  countLocations!: Sequelize.BelongsToManyCountAssociationsMixin;
  // DataSource hasMany Measurement via DataSourceId
  Measurements!: Measurement[];
  getMeasurements!: Sequelize.HasManyGetAssociationsMixin<Measurement>;
  setMeasurements!: Sequelize.HasManySetAssociationsMixin<Measurement, MeasurementId>;
  addMeasurement!: Sequelize.HasManyAddAssociationMixin<Measurement, MeasurementId>;
  addMeasurements!: Sequelize.HasManyAddAssociationsMixin<Measurement, MeasurementId>;
  createMeasurement!: Sequelize.HasManyCreateAssociationMixin<Measurement>;
  removeMeasurement!: Sequelize.HasManyRemoveAssociationMixin<Measurement, MeasurementId>;
  removeMeasurements!: Sequelize.HasManyRemoveAssociationsMixin<Measurement, MeasurementId>;
  hasMeasurement!: Sequelize.HasManyHasAssociationMixin<Measurement, MeasurementId>;
  hasMeasurements!: Sequelize.HasManyHasAssociationsMixin<Measurement, MeasurementId>;
  countMeasurements!: Sequelize.HasManyCountAssociationsMixin;
  // DataSource belongsTo Location via LocationId
  Location!: Location;
  getLocation!: Sequelize.BelongsToGetAssociationMixin<Location>;
  setLocation!: Sequelize.BelongsToSetAssociationMixin<Location, LocationId>;
  createLocation!: Sequelize.BelongsToCreateAssociationMixin<Location>;

  static initModel(sequelize: Sequelize.Sequelize): typeof DataSource {
    DataSource.init({
    DataSourceId: {
      autoIncrement: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    DataSourceUri: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: "DataSource_DataSourceUri_LocationId_key"
    },
    LocationId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'Location',
        key: 'LocationId'
      },
      unique: "DataSource_DataSourceUri_LocationId_key"
    }
  }, {
    sequelize,
    tableName: 'DataSource',
    schema: 'public',
    timestamps: false,
    indexes: [
      {
        name: "DataSource_DataSourceUri_LocationId_key",
        unique: true,
        fields: [
          { name: "DataSourceUri" },
          { name: "LocationId" },
        ]
      },
      {
        name: "DataSource_DataSourceUri_key",
        unique: true,
        fields: [
          { name: "DataSourceUri" },
        ]
      },
      {
        name: "DataSource_pkey",
        unique: true,
        fields: [
          { name: "DataSourceId" },
        ]
      },
    ]
  });
  return DataSource;
  }
}

```
![database-model](https://user-images.githubusercontent.com/35292572/118113710-e887dc80-b3e6-11eb-99c2-40a307f768bd.png)

**Comment by steveschmitt** [comment]
Thanks.  I think sequelize-auto is treating Measurement as a many-to-many junction table, since it has keys for both DataSource and Location.  Maybe sequelize-auto should decide it is a junction table only if the the PK has exactly two columns.

You could just ignore those extra properties in the TypeScript, but what's this about a `duplicate identifier` error?

**Comment by alexanderdavide** [comment]
I see. This would solve this issue but wouldn't this bring up other problems?

Can you please elaborate on "ignore those extra properties in the TypeScript"? I guess you want to say that I should just not use or delete them. The problem is the `duplicate identifier` errors. If you look at the classes, you can see that some methods are duplicated due to the wrong relationships. The according error is `duplicate identifier`. Of course, I need a reliable generation process. Deleting the wrong parts every time I regenerate the models is not viable.

**Comment by steveschmitt** [comment]
Fixed in 0.8.3
